### PR TITLE
pkp/pkp-lib#8290 Inclusion of 'Data Availability Statements' as submission metadata

### PR DESCRIPTION
**Pull request by iudizm**
Related issue: #8290

### CODE REVIEW

File: schemas/context.json
Comment (asmecher) [inline schemas/context.json:847]: Sorry, I know this is trivial, but can you format the JSON changes with tab-based indentation? The changes in this PR use a mix of tabs and spaces.

File: templates/controllers/modals/submission/viewSubmissionMetadata.tpl
Comment (asmecher) [inline templates/controllers/modals/submission/viewSubmissionMetadata.tpl:25]: It looks like you've identified missing escaping here -- but if so, can you add it to the above `<th ...>` element as well?

File: schemas/context.json
Comment (iudizm) [reply schemas/context.json]: Sure, sorry for mixing indentation styles

File: templates/controllers/modals/submission/viewSubmissionMetadata.tpl
Comment (iudizm) [reply templates/controllers/modals/submission/viewSubmissionMetadata.tpl:25]: done, escaping the `<th ...>` as well now

**Comment by iudizm** [comment]
@asmecher, done implementing the Cypress test for the metadata, could you take a look?

note: 
You said the test should "check an existing review assignment to ensure that it's visible", but there wasn't any review assignments with the 'disclosed author' method in the [dataset](https://github.com/pkp/datasets/tree/main/ojs/main/mysql), so I had to [create a new assignment](https://github.com/pkp/pkp-lib/pull/8291/files#diff-bb23160115233ba107b181077d37776cd48b6b08bca81dc34a86f0a99c54cacdR29) to check out the presence of the metadata in this type of review.
I hope this is the right way to do it.

File: cypress/tests/integration/DataAvailabilityStatements.spec.js
Comment (asmecher) [inline cypress/tests/integration/DataAvailabilityStatements.spec.js:20]: It's best not to embed database IDs in the test. The best approach is to have the test find the content e.g. via the editorial interface by title or author name, then store its ID in an [alias](https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Aliases) for the testing code to use.

Admittedly, sometimes we do embed database IDs in our own tests; if you do continue that way, it's best to at least pull this out into a constant near the top of the file.

File: cypress/tests/integration/DataAvailabilityStatements.spec.js
Comment (asmecher) [inline cypress/tests/integration/DataAvailabilityStatements.spec.js:32]: `cy.wait` statements [should be considered FIXMEs](https://docs.cypress.io/guides/references/best-practices#Unnecessary-Waiting). We do have a few cases where we use waits internally -- also needing to be fixed e.g. due to some sort of Javascript rewrite -- but they shouldn't be used indiscriminately or it'll lead to balancing random beakage of the tests against the whole suite taking too long to run.

(For example, there is a 5.5 second delay just below -- that's a lot of time for the server to sit around!)

File: schemas/publication.json
Comment (asmecher) [inline schemas/publication.json:333]: The indentation contains a mix of spaces and tabs, here and elsewhere in the PR; please follow the existing conventions for each file. (We're gradually moving to autoformatting with git hooks, but that hasn't happened everywhere yet.)

**Review by asmecher — CHANGES_REQUESTED** [review]
Sorry for the delay in getting around to a second review! I have added a couple of comments on the tests, plus one in the OJS PR that may affect the other application PRs as well.

File: cypress/tests/integration/DataAvailabilityStatements.spec.js
Comment (iudizm) [reply cypress/tests/integration/DataAvailabilityStatements.spec.js]: Sorry for this smelly code, done removing the database IDs from the specs

File: cypress/tests/integration/DataAvailabilityStatements.spec.js
Comment (iudizm) [reply cypress/tests/integration/DataAvailabilityStatements.spec.js]: Done, sorry for this bad practice

**Comment by iudizm** [comment]
@asmecher,

We moved the test about the metadata in the Review to the OJS specific integration specs. 
So it is now [here](https://github.com/pkp/ojs/pull/3553/files).

**Comment by NateWr** [comment]
Replaced by https://github.com/pkp/pkp-lib/pull/8570

**Pull request closed without merging**
